### PR TITLE
Emit Sync Job Finished Status After Sync Completion

### PR DESCRIPTION
**Pull request by ndegwamartin**
**Description**
@jingtang10 @aditya-07 I think we should emit the _Success status_ as well. Currently once its done downloading we don't notify dependent apps.

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by omarismail94 — APPROVED** [review]
@ndegwamartin please update your PR with master! Ive approved it, but it needs to be up to date with master

File: engine/src/main/java/com/google/android/fhir/sync/FhirSynchronizer.kt [hunk 1]
```diff
@@ -116,6 +116,7 @@ internal class FhirSynchronizer(
       }
     }
     return if (exceptions.isEmpty()) {
+      setSyncState(SyncJobStatus.Finished())
```
Comment (aditya-07) [inline engine/src/main/java/com/google/android/fhir/sync/FhirSynchronizer.kt:119]: `SyncJobStatus.Finished` signifies that the sync has completed/finished successfully and this change is setting it early irrespective of whether `upload` is successful or not.
That's why [FhirSynchronizer.kt#L91]( https://github.com/google/android-fhir/blob/a0e915bd01eaeb605685fb0295d93fa00bd6563c/engine/src/main/java/com/google/android/fhir/sync/FhirSynchronizer.kt#L91 ) is the correct place to set the overall success status.

If we need the individual success status for upload and download, we can maybe think of adding new status values.

**Review by aditya-07 — CHANGES_REQUESTED** [review]

**Review by omarismail94 — COMMENTED** [review]
Good catch @aditya-07 ! Yes, this is correct

**Comment by ndegwamartin** [comment]
> Good catch @aditya-07 ! Yes, this is correct

Noted

**Comment by ndegwamartin** [comment]
Closing this for now

**Pull request closed without merging**